### PR TITLE
Add video capture properties and set preferred api backend

### PR DESCRIPTION
**Pull request by JoshChristie**
- Add missing VideoCaptureProperties
- Add ability to set the preferred api backend

### CODE REVIEW

**Review by OmriTreidel — APPROVED** [review]
looks great. For some reason The build is failing

**Comment by JoshChristie** [comment]
Fixed the build.

File: videoio_test.go [hunk 1]
```diff
@@ -91,12 +128,53 @@ func TestVideoCaptureFile(t *testing.T) {
...
 
+	// video capture file with non-existent video
 	vc2, err := VideoCaptureFile("nonexistent.mp4")
 	defer vc2.Close()
 
 	if err == nil {
 		t.Errorf("Expected error when opening invalid file")
 	}
+
+	t.Run(" video capture file with api", func(t *testing.T) {
```
Comment (deadprogram) [inline videoio_test.go:139]: Note extra space.

**Comment by deadprogram** [comment]
Now going to squash/merge this PR as it has all the needed requirements.

Thank you very much for this contribution @JoshChristie

**Pull request merged**